azext_edge/tests/edge/support/conftest.py

```diff
@@ -31,17 +31,21 @@ def add_pod_to_mocked_pods(
         spec = V1PodSpec(containers=[V1Container(name=container_name)])
         pod = V1Pod(metadata=V1ObjectMeta(namespace=namespace, name=pod_name), spec=spec)
 
-        if mock_init_containers:
+        if mock_init_container:
             pod.spec.init_containers = [V1Container(name="mock-init-container")]
-            expected_pod_map[namespace][pod_name]["mock-init-container"] = mock_log
+            expected_pod_map[namespace][pod_name] = {"mock-init-container": mock_log}
         pod_list.append(pod)
-        expected_pod_map[namespace][pod_name] = {container_name: mock_log}
+
+        if pod_name in expected_pod_map[namespace]:
+            expected_pod_map[namespace][pod_name].update({container_name: mock_log})
+        else:
+            expected_pod_map[namespace][pod_name] = {container_name: mock_log}
 
     pods_list = V1PodList(items=pod_list)
     mocked_client.CoreV1Api().list_pod_for_all_namespaces.return_value = pods_list
     mocked_client.CoreV1Api().read_namespaced_pod_log.return_value = mock_log
 
-    yield expected_pod_map
+    return expected_pod_map
```

Comment:
oh I see, even deleting the return will have the same effect to auto apply the update to mock_pod_list?

Reply:
since you are modifying the object - yes it will. If you want to google a term, search fruitful functions (idk what the alternative is called - void?)

Reply:
looks like it works! will update thanks!